src/main/java/chat/command/Command.java

```diff
@@ -0,0 +1,70 @@
+package chat.command;
+
+import chat.ChatException;
+import chat.TaskList;
+import chat.Storage;
+import chat.Ui;
+import chat.task.Task;
+
+/**
+ * Abstract class for commands.
+ * <p>Commands refer to the commands that Chat the Cat receives from the user.</p>
+ */
+
+public abstract class Command {
+
+    /**
+     * Abstract method that executes the command.
+     * @param tasks TaskList object that contains a list of tasks.
+     * @param ui Ui object that gives responses to user.
+     * @param storage Storage object that interacts with task data in hard disk.
+     * @throws ChatException If format of command is wrong.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws ChatException;
+
+    /**
+     * Returns true if command requires Chat the Cat to exit.
+     * 
+     * @return Boolean that shows if Chat should exit.
+     */
+    public boolean isExit() { 
+        return false;
+    };
+
+    /**
+     * Method that can be used to check format for commands with indexes.
+     * 
+     * @param tasks TaskList object that contains a list of tasks.
+     * @param commandName Command name.
+     * @param str Inputted command string from user to Chat the Cat.
+     * @return Task.
+     * @throws ChatException If format of command is wrong.
+     */
+    public Task checkCommandIndex(TaskList tasks, String commandName, String str) throws ChatException {
+        //for commands with format: [command] [index]
+        //returns valid index if index is correct
```

Comment:
the structure here is abit hard to follow... since _"Please input with this ... "_ occurs quite alot in the exceptions, maybe can refactor that phrase out of all the exceptions which point to formatting error?

Reply:
Yes I think that this method could be made cleaner, thanks for the suggestion.